packages/typespec-client-generator-core/src/package.ts

```diff
@@ -413,78 +414,113 @@ function getSdkMethods<TServiceOperation extends SdkServiceOperation>(
   return diagnostics.wrap(retval);
 }
 
+function getEndpointTypeFromSingleServer(
+  context: TCGCContext,
+  client: SdkClient | SdkOperationGroup,
+  server: HttpServer | undefined
+): [SdkEndpointType[], readonly Diagnostic[]] {
+  const diagnostics = createDiagnosticCollector();
+  const templateArguments: SdkPathParameter[] = [];
+  const defaultOverridableEndpointType: SdkEndpointType = {
+    kind: "endpoint",
+    serverUrl: "{endpoint}",
+    templateArguments: [
+      {
+        name: "endpoint",
+        isGeneratedName: true,
+        description: "Service host",
+        kind: "path",
+        onClient: true,
+        urlEncode: false,
+        optional: false,
```

Comment:
it seems all the endpoint param is not optional? then param with default value is also not optional, is it by design?

Reply:
yeah, i realized the endpoint is always required to be sent to the service: there might be a client default value, but that client default value still needs to be sent